index.js

```diff
@@ -0,0 +1,242 @@
+//import libraries
+const dotenv = require("dotenv").config();
+const phoenix = require("@phoenixlan/phoenix.js");
+const amqp = require("amqplib/callback_api");
+const {
+    UTCDate
+} = require("@date-fns/utc");
+const {
+    addHours,
+    subMonths,
+    differenceInMilliseconds,
+    differenceInHours
+} = require("date-fns");
+const {
+    Client,
+    GatewayIntentBits,
+    EmbedBuilder,
+    Guild
+} = require("discord.js");
+//initialise the api
+phoenix.init(process.env.INIT_URL);
+//change this to token from login in the api
+phoenix.User.Oauth.setAuthState(process.env.TOKEN, process.env.REFRESH_TOKEN);
+//variable that controls the months before the next event crew roles should be removed
+const timeBeforeNextEventRemove = process.env.REMOVE_MONTHS;
+//change this, guild id to ensure proper guild selection
+let phoenixGuildId = process.env.GUILD_ID;
+//set the discord token variable with the token from .env file
+const DISCORD_TOKEN = process.env.BOT_TOKEN;
+//change this to the rabbitmq address and port
+amqp.connect(process.env.RABBITMQ_CONNECT_LINK, function (error0, connection) {
+    if (error0) {
+        throw error0;
+    }
+    connection.createChannel(function (error1, channel) {
+        if (error1) {
+            throw error1;
+        }
+        //change this
+        let queue = "";
+
+
+        channel.assertQueue(queue, {
+            durable: true,
+        });
+
+        channel.consume(queue, async function (msg) {
+            console.log(msg.content.toString());
+            removeAllRoles();
+            updateRoles();
+        }, {
+            noAck: true,
+        });
+
+    });
+});
+const phoenixClient = new Client({
+    intents: [
+        GatewayIntentBits.Guilds,
+        GatewayIntentBits.GuildMessages,
+        GatewayIntentBits.MessageContent,
+        GatewayIntentBits.GuildMembers,
+    ],
+});
+
+async function handleRoleRemoval() {
+    const now = new UTCDate();
+    const events = await phoenix.getEvents();
+    const nextEventTimeOffsetted = events
+        .map(event => subMonths(new UTCDate(event["start_time"] * 1000), timeBeforeNextEventRemove))
+        .find(time => time.getTime() > now.getTime());
+    if (!nextEventTimeOffsetted || differenceInHours(nextEventTimeOffsetted, now) > 25) {
+        console.info("Skipping scheduled role removal");
+        return;
+    }
+
+    console.log("Removing roles");
+    removeAllRoles();
+}
+setInterval(() => {
+    handleRoleRemoval();
+}, 8.64e+7); // 1 day
```

Comment:
please use proper math here, i.e 60*60*24. Or use the whole number. Using an exponential here is probably a bad practice.